src/Serialization.purs

```diff
@@ -193,45 +142,30 @@ convertTxOutputs arrOutputs = do
 
 convertTxOutput :: T.TransactionOutput -> Effect TransactionOutput
 convertTxOutput (T.TransactionOutput { address, amount, data_hash }) = do
-  address' <- convertAddress address
   value <- convertValue amount
-  txo <- newTransactionOutput address' value
+  txo <- newTransactionOutput address value
   for_ (unwrap <$> data_hash) \bytes -> do
     for_ (fromBytes bytes) $
       transactionOutputSetDataHash txo
   pure txo
 
-convertAddress :: T.Address -> Effect Address
-convertAddress address = do
-  let baseAddress = unwrap (unwrap address)."AddrType"
-  payment <- case baseAddress.payment of
-    T.PaymentCredentialKey (T.Ed25519KeyHash keyHashBytes) -> do
-      newStakeCredentialFromKeyHash =<< fromBytesEffect keyHashBytes
-    T.PaymentCredentialScript (T.ScriptHash scriptHashBytes) -> do
-      newStakeCredentialFromScriptHash =<< fromBytesEffect scriptHashBytes
-  stake <- case baseAddress.stake of
-    T.StakeCredentialKey (T.Ed25519KeyHash keyHashBytes) -> do
-      newStakeCredentialFromKeyHash =<< fromBytesEffect keyHashBytes
-    T.StakeCredentialScript (T.ScriptHash scriptHashBytes) -> do
-      newStakeCredentialFromScriptHash =<< fromBytesEffect scriptHashBytes
-  base_address <- newBaseAddress baseAddress.network payment stake
-  baseAddressToAddress base_address
-
 convertValue :: Value.Value -> Effect Value
 convertValue val = do
   let
     lovelace = Value.valueToCoin' val
     m = Value.getNonAdaAsset' val
   multiasset <- newMultiAsset
   forWithIndex_ m \scriptHashBytes' values -> do
-    let scriptHashBytes = Value.getCurrencySymbol scriptHashBytes'
+    let mScripthash = scriptHashFromBytes $ Value.getCurrencySymbol scriptHashBytes'
     assets <- newAssets
+
+    scripthash <- errorOnNothing mScripthash
```

Comment:
Let's use `explain` in cases like this, if we ever get a bug report, a message can save us a lot of time.